From 88d96032c24f547c77cf1458a2f885967c18ca7d Mon Sep 17 00:00:00 2001
From: Michael Booth <m.judd.booth@gmail.com>
Date: Sat, 23 Aug 2014 22:42:02 -0400
Subject: [PATCH 0278/1009] Make the error explain what datatype it's talking
 about

---
 pandas/tools/merge.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pandas/tools/merge.py b/pandas/tools/merge.py
index d17e2e2..39bddeb 100644
--- a/pandas/tools/merge.py
+++ b/pandas/tools/merge.py
@@ -765,7 +765,7 @@ class _Concatenator(object):
         ndims = set()
         for obj in objs:
             if not isinstance(obj, NDFrame):
-                raise TypeError("cannot concatenate a non-NDFrame object")
+                raise TypeError("cannot concatenate a non-NDFrame object {}".format(type(obj)))
 
             # consolidate
             obj.consolidate(inplace=True)
-- 
2.6.0

