From 935b02bbee0fd2a3238eae31fc8c0c5757e8af32 Mon Sep 17 00:00:00 2001
From: Michael Booth <m.judd.booth@gmail.com>
Date: Wed, 8 Apr 2015 20:24:42 -0400
Subject: [PATCH 1003/1009] comment on tz in holiday dates.

---
 pandas/tseries/holiday.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pandas/tseries/holiday.py b/pandas/tseries/holiday.py
index 3b3542b..1826f34 100644
--- a/pandas/tseries/holiday.py
+++ b/pandas/tseries/holiday.py
@@ -203,6 +203,8 @@ class Holiday(object):
         end_date = Timestamp(end_date)
 
         year_offset = DateOffset(years=1)
+        #FIXME: Why does original code ignore the timezone of start_date?.  propose this:
+        # base_date = Timestamp(datetime(start_date.year, self.month, self.day), tz = start_date.tz)
         base_date = Timestamp(datetime(start_date.year, self.month, self.day))
         dates = DatetimeIndex(start=base_date, end=end_date, freq=year_offset)
         holiday_dates = self._apply_rule(dates)
-- 
2.6.0

